apps/apps.c

```diff
@@ -2343,6 +2349,117 @@ int app_isdir(const char *name)
 }
 #endif
 
+/* app_dirname section */
+
+/*
```

Comment:
I really really dislike this.  This should only be used as a replacement for dirname on places where that doesn't exist.  I.e., non-posix, right?  And also the name "fallback" is not common and I find it jarring.

Reply:
Who cares if it isn't common?  This is apps and it's a local function, and that function *does* work as a fallback.  How would you name it otherwise?

Reply:
app_dirname, like app_malloc etc

Reply:
If you scroll down, you'll see an `app_dirname` for Unix and an `app_dirname` for VMS.  Point is that for VMS, it's possible to use a Unix like file spec syntax with some applications, and I thought openssl could be one of them.  So that means that when the classic VMS syntax scan fails, it *falls back* to trying the Unix syntax as well.  So `fallback_dirname` is a common function for more than one platform, and of course, the Unix variant simply calls `fallback_dirname` directly.

Reply:
then maybe posix_dirname?

Reply:
Sure, if that feels better

Reply:
Strictly speaking, this comes from #3709.  Sure, it was my suggestion, but...